woocommerce/google-listings-and-ads
# Use @emotion library for CSS

### Changes proposed in this Pull Request:

Closes #65 .

This PR looks into using [emotion library](https://emotion.sh/docs/introduction) to do CSS in JS. Emotion is used in gutenberg.

Some findings from this research:

- `@emotion/css` is framework-agnostic, you can use it in HTML with vanilla JS, no react required.
- `@emotion/react` is react specific.
- `@emotion/styled` is similar to [styled-components](https://styled-components.com/) library. It has a peer dependency on `@emotion/react`, so you have to manually install `@emotion/react` even though you are not using it explicitly.
- `@emotion/react` and `@emotion/css` both have a `css` function but **they are not the same**! This gave me a gotcha headache. This is briefly mentioned here: https://emotion.sh/docs/package-summary
- On the official site, it is recommended to use `@emotion/react` for react project, however I find it generate more noise and more tedious because you either have to modify babel config or put in extra JSX Pragma code at the top of the your JS files. More info here: https://emotion.sh/docs/css-prop . Because of this, I personally do not favor `@emotion/react`.
- We can use `@emotion/styled` to create styled components. An example here: https://github.com/woocommerce/google-listings-and-ads/commit/239d1a3faf3836273479bf80fbf3614f438484cf . I feel this is ok, but my experience tells me we might end up having a lot of `StyledXyz` component and can be quite a bit of noise and mentally taxing to comprehend and navigate.
- I'm using `@emotion/css` library here in this PR. Essentially it generates unique class names that we can use, without having to worry about class name conflict. You can see it's pretty much similar with previous SCSS approach. I think this is a more balanced approach and more welcoming to people who are not familiar.
- Note that we could have also changed all the manual custom CSS classes (e.g. `motivation-text`, `title`, `description` etc) to use the `css` function to auto-generate unique class names, so that they wouldn't be affected by CSS from other places (e.g. defined and clashed in global style sheet). I leave them out for now.
- Because this is essentially JavaScript, an advantage over the SCSS approach is that we can make good use of editor JS support such as "rename variable" and "find all references". i.e. less spelling mistake error.
- Install [vscode-styled-components extension](https://marketplace.visualstudio.com/items?itemName=jpoissonnier.vscode-styled-components) to have better syntax highlighting for CSS in JS in your VS Code editor!

### Detailed test instructions:

1. Checkout the code.
2. `npm install` and `npm run start`.
3. Go to http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart and make sure everything still looks ok and same as before.

### Changelog Note:

Use emotion library for styling and CSS.
